codec/src/main/java/io/netty/handler/codec/DefaultTextHeaders.java

```diff
@@ -396,4 +404,226 @@ public TextHeaders clear() {
+        }
+
+        @Override
+        public TextHeaders set(CharSequence name, CharSequence... values) {
+            DefaultTextHeaders.super.set(name, commaSeparate(charSequenceEscaper, values));
+            return DefaultTextHeaders.this;
+        }
+
+        @Override
+        public TextHeaders set(CharSequence name, Iterable<? extends CharSequence> values) {
+            DefaultTextHeaders.super.set(name, commaSeparate(charSequenceEscaper, values));
+            return DefaultTextHeaders.this;
+        }
+
+        @Override
+        public TextHeaders setObject(CharSequence name, Object... values) {
+            DefaultTextHeaders.super.set(name, commaSeparate(objectEscaper, values));
+            return DefaultTextHeaders.this;
+        }
+
+        @Override
+        public TextHeaders setObject(CharSequence name, Iterable<?> values) {
+            DefaultTextHeaders.super.set(name, commaSeparate(objectEscaper, values));
+            return DefaultTextHeaders.this;
+        }
+
+        private TextHeaders addEscapedValue(CharSequence name, CharSequence escapedValue) {
+            CharSequence currentValue = DefaultTextHeaders.super.get(name);
+            if (currentValue == null) {
+                DefaultTextHeaders.super.add(name, escapedValue);
+            } else {
+                DefaultTextHeaders.super.set(name, commaSeparateEscapedValues(currentValue, escapedValue));
+            }
+            return DefaultTextHeaders.this;
+        }
+
+        private <T> CharSequence commaSeparate(ValueEscaper<T> escaper, T... values) {
+            StringBuilder sb = new StringBuilder(values.length * DEFAULT_VALUE_SIZE);
+            if (values.length > 0) {
+                int end = values.length - 1;
+                for (int i = 0; i < end; i++) {
+                    escaper.escape(sb, values[i]).append(", ");
+                }
+                escaper.escape(sb, values[end]);
+            }
+            return sb.toString();
+        }
+
+        private <T> CharSequence commaSeparate(ValueEscaper<T> escaper, Iterable<? extends T> values) {
+            StringBuilder sb = new StringBuilder();
+            Iterator<? extends T> iterator = values.iterator();
+            if (iterator.hasNext()) {
+                T next = iterator.next();
+                while (iterator.hasNext()) {
+                    escaper.escape(sb, next).append(", ");
+                    next = iterator.next();
+                }
+                escaper.escape(sb, next);
+            }
+            return sb.toString();
+        }
+
+        private CharSequence commaSeparateEscapedValues(CharSequence currentValue, CharSequence value) {
+            return new StringBuilder(currentValue.length() + 2 + value.length())
+                    .append(currentValue)
+                    .append(", ")
+                    .append(value).toString();
+        }
+
+        private StringBuilder quote(StringBuilder sb, CharSequence value) {
```

Comment:
We should also add a unit test that adds the same value again and check that the result is the same:

```java
headers.add("foo", "some, value, with stuff \"");
String value = headers.get("foo");
// Possibly do the following 2 lines in a loop
headers.add("foo", value);
assertEquals(value, headers.get("foo"));
```